File: Python/Calculadora.py
import time

def suma(a, b):
    return a + b

def resta(a, b):
    return a - b

def multiplicacion(a, b):
    return a * b

def division(a, b):
    if b != 0:
        return a / b
    else:
        return "No se puede dividir por cero"

while True:
    print("\nSeleccione la operación que desea realizar:")
    print("1. Suma")
    print("2. Resta")
    print("3. Multiplicación")
    print("4. División")
    print("5. Salir")

    opcion = input("Ingrese el número de la operación: ")

    if opcion == '5':
        print("Saliendo del programa.")
        break

    if opcion in ['1', '2', '3', '4']:
        num1 = int(input("Ingrese el primer número: "))
        num2 = int(input("Ingrese el segundo número: "))

        start_time = time.time()  # Start timing the operation

        if opcion == '1':
            resultado = suma(num1, num2)
            print("Resultado de la suma:", resultado)
        elif opcion == '2':
            resultado = resta(num1, num2)
            print("Resultado de la resta:", resultado)
        elif opcion == '3':
            resultado = multiplicacion(num1, num2)
            print("Resultado de la multiplicación:", resultado)
        elif opcion == '4':
            resultado = division(num1, num2)
            print("Resultado de la división:", resultado)

        end_time = time.time()  # End timing the operation
        execution_time = end_time - start_time
        print(f"Tiempo de ejecución de la operación: {execution_time:.6f} segundos")
    else:
        print("Opción no válida. Por favor, intente de nuevo.")

File: Calculadora.java
import java.math.BigInteger;
import java.util.Scanner;

public class Operaciones {

    public static void main(String[] args) {
    	
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Ingrese el primer número: ");
        String input1 = scanner.nextLine();
        BigInteger numero1 = new BigInteger(input1);

        System.out.print("Ingrese el segundo número: ");
        String input2 = scanner.nextLine();
        BigInteger numero2 = new BigInteger(input2);

        System.out.println("\nSeleccione la operación que desea realizar:");
        System.out.println("1. Suma");
        System.out.println("2. Resta");
        System.out.println("3. Multiplicación");
        System.out.println("4. División");
        System.out.print("Ingrese el número de la operación: ");
        
        int opcion = scanner.nextInt();
        
  
        long startTime = System.currentTimeMillis();

        
        BigInteger resultado = null;

        
        switch (opcion) {
            case 1:
                resultado = numero1.add(numero2);
                System.out.println("Resultado de la suma: " + resultado);
                break;
            case 2:
                resultado = numero1.subtract(numero2);
                System.out.println("Resultado de la resta: " + resultado);
                break;
            case 3:
                resultado = numero1.multiply(numero2);
                System.out.println("Resultado de la multiplicación: " + resultado);
                break;
            case 4:
                if (!numero2.equals(BigInteger.ZERO)) {
                    resultado = numero1.divide(numero2);
                    System.out.println("Resultado de la división: " + resultado);
                } else {
                    System.out.println("División: No se puede dividir por cero.");
                }
                break;
            default:
                System.out.println("Opción no válida.");
                break;
        }

        
        long endTime = System.currentTimeMillis();
        long executionTime = endTime - startTime;
        System.out.println("Tiempo de ejecución: " + executionTime + " ms");

       
        scanner.close();
    }
}
